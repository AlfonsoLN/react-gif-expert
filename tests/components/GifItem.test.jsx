import { render, screen } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

describe('Prueba en <GifItem />', () => {
    const titulo = 'Hola Mundo';
    const direccion = 'https://example.com/image.jpg';

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={titulo} url={direccion}/>);
                
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={titulo} url={direccion}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(direccion);
        // expect(screen.getByRole('img').alt).toBe(titulo);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(direccion);
        expect(alt).toBe(titulo);
    });

    test('debe mostrar el tÃ­tulo en el componente', () => {
        render(<GifItem title={titulo} url={direccion}/>);

        expect(screen.getByText(titulo)).toBeTruthy();
    });
});