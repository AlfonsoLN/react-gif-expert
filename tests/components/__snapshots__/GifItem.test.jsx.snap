// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola Mundo"
      src="https://example.com/image.jpg"
    />
    <p>
      Hola Mundo
    </p>
  </div>
</div>
`;
